warehouse/query-core/src/main/java/datawave/query/tld/TLDIndexBuildingVisitor.java

```diff
@@ -233,7 +233,7 @@ public Range getKeyRange(Entry<Key,Document> from) {
             
             @Override
             public EventDataQueryFilter clone() {
-                return this.clone();
```

Comment:
IL_INFINITE_RECURSIVE_LOOP

Reply:
Hmm, not familiar enough with this part of the code, but having clone not actually clone doesn't seem great to me.  Maybe @FineAndDandy knows?

Reply:
Yeah, I assumed that cloning wasn't really happening for these objects, otherwise we'd have run into OOM at some point.

Reply:
To your point though, perhaps throwing **UnsupportedOperationException** is the better option in these cases